hydromt/data_catalog.py

```diff
@@ -1286,7 +1291,7 @@ def get_rasterdataset(
         provider: Optional[str] = None,
         version: Optional[str] = None,
         **kwargs,
-    ) -> xr.Dataset:
+    ) -> Optional[xr.Dataset]:
         """Return a clipped, sliced and unified RasterDataset.
 
         To clip the data to the area of interest, provide a `bbox` or `geom`,
```

Comment:
I think it good in this docstring (and all other get_* methods in this file) to mention that this function returns either data, error if RAISE and None if IGNORE. This is the main doc for the users.